substrate/pkg/controller/substrate/controller.go

```diff
@@ -100,6 +100,7 @@ func (c *Controller) Reconcile(ctx context.Context, substrate *v1alpha1.Substrat
 			}
 			result, err := f(ctx, mutable)
 			if err != nil {
+				logging.FromContext(ctx).Debugf("Reconciling %s, error %v", reflect.ValueOf(resource).Elem().Type(), err)
```

Comment:
Why do we need to log error here? If this isbubbled up at the end w/ the multierr.Combine, do we log the same error twice?

Reply:
In some cases, we just get stuck in a for loop without exiting. This is to catch such scenarios. Recent scenario we ran into was when AWS_REGION is not set. Now that we have this `for ctx.Err() == nil`, don't need this log anymore. I will remove it.